sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -0,0 +1,630 @@
+#endif  /* GNRC_IPV6_NIB_CONF_6LN */
+#if GNRC_IPV6_NIB_CONF_MULTIHOP_P6C
+        case GNRC_IPV6_NIB_ABR_TIMEOUT:
+            /* TODO */
+            break;
+#endif  /* GNRC_IPV6_NIB_CONF_MULTIHOP_P6C */
+        case GNRC_IPV6_NIB_PFX_TIMEOUT:
+            /* TODO */
+            break;
+        case GNRC_IPV6_NIB_RTR_TIMEOUT:
+            /* TODO */
+            break;
+        default:
+            break;
+    }
+    mutex_unlock(&_nib_mutex);
+}
+
+/* Iterator for NDP options in a packet */
+#define FOREACH_OPT(ndp_pkt, opt, icmpv6_len) \
+    for (opt = (ndp_opt_t *)(ndp_pkt + 1); \
+         icmpv6_len > 0; \
+         icmpv6_len -= (opt->len << 3), \
+         opt = (ndp_opt_t *)(((uint8_t *)opt) + (opt->len << 3)))
+
+static size_t _get_l2src(kernel_pid_t iface, uint8_t *l2src,
+                         size_t l2src_maxlen)
+{
+    bool try_long = false;
+    int res;
+    uint16_t l2src_len;
+    /* maximum address length that fits into a minimum length (8) S/TL2A
+     * option */
+    const uint16_t max_short_len = 6;
+
+    /* try getting source address */
+    if ((gnrc_netapi_get(iface, NETOPT_SRC_LEN, 0, &l2src_len,
+                         sizeof(l2src_len)) >= 0) &&
+        (l2src_len > max_short_len)) {
+        try_long = true;
+    }
+
+    if (try_long && ((res = gnrc_netapi_get(iface, NETOPT_ADDRESS_LONG, 0,
+                                            l2src, l2src_maxlen)) > max_short_len)) {
+        l2src_len = (uint16_t)res;
+    }
+    else if ((res = gnrc_netapi_get(iface, NETOPT_ADDRESS, 0, l2src,
+                                    l2src_maxlen)) >= 0) {
+        l2src_len = (uint16_t)res;
+    }
+    else {
+        DEBUG("nib: No link-layer address found.\n");
+        l2src_len = 0;
+    }
+
+    return l2src_len;
+}
+
+static void _send_delayed_nbr_adv(const gnrc_ipv6_netif_t *netif,
+                                  const ipv6_addr_t *tgt,
+                                  const ipv6_addr_t *dst,
+                                  gnrc_pktsnip_t *reply_aro)
+{
+    gnrc_pktsnip_t *nbr_adv, *extra_opts = reply_aro;
+    _nib_onl_entry_t *nce;
+    uint8_t reply_flags = NDP_NBR_ADV_FLAGS_S;
+
+#if GNRC_IPV6_NIB_CONF_ROUTER
+    if (_is_rtr(netif)) {
+        reply_flags |= NDP_NBR_ADV_FLAGS_R;
+    }
+#endif
+    if (ipv6_addr_is_multicast(dst)) {
+        uint8_t l2addr[GNRC_IPV6_NIB_L2ADDR_MAX_LEN];
+        size_t l2addr_len = _get_l2src(netif->pid, l2addr, sizeof(l2addr));
+        if (l2addr_len > 0) {
+            extra_opts = gnrc_ndp2_opt_tl2a_build(l2addr, l2addr_len,
+                                                  extra_opts);
+            if (extra_opts == NULL) {
+                DEBUG("nib: No space left in packet buffer. Not replying NS");
+                gnrc_pktbuf_release(reply_aro);
+                return;
+            }
+        }
+        else {
+            reply_flags |= NDP_NBR_ADV_FLAGS_O;
+        }
+    }
+    else {
+        reply_flags |= NDP_NBR_ADV_FLAGS_O;
+    }
```

Comment:
```c
        else {
            reply_flags |= NDP_NBR_ADV_FLAGS_O;
        }
    }
    else {
        reply_flags |= NDP_NBR_ADV_FLAGS_O;
    }
```

you were able to remove such code duplication in another PR, can you also apply this here, please?

Reply:
There it was at the end of a function, so I was just able to early exit. This is not possible here.

Reply:
I could use a "hidden goto" instead ;-)